org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/concurrent/AbstractRequest.java

```diff
@@ -34,19 +49,42 @@
 
 	protected AbstractRequest(RequestManager requestManager) {
 		this.requestManager = requestManager;
-		this.result = new CompletableFuture<>();
-		this.cancelIndicator = new RequestCancelIndicator(this.result);
+		this.result = new ResultFuture();
+		this.cancelIndicator = new RequestCancelIndicator(this);
+	}
+
+	void cancelResult() {
```

Comment:
package private by intention?

Reply:
Yes, it was intentionally - but only for the local code. I'll make it protected.